aptos-move/framework/aptos-framework/sources/BigVector.move

```diff
@@ -25,40 +31,59 @@ module AptosFramework::BigVector {
         assert!(bucket_size > 0, 0);
         BigVector {
             buckets: Table::new(),
-            next_index: BigVectorIndex {
+            end_index: BigVectorIndex {
                 bucket_index: 0,
                 vec_index: 0,
             },
+            num_buckets: 0,
             bucket_size,
         }
     }
 
+    /// Create an empty vector with `num_buckets` reserved.
+    public fun new_with_capacity<T: store>(bucket_size: u64, num_buckets: u64): BigVector<T> {
+        let v = new(bucket_size);
+        reserve(&mut v, num_buckets);
+        v
+    }
+
     /// Destroy the vector `v`.
     /// Aborts if `v` is not empty.
     public fun destroy_empty<T>(v: BigVector<T>) {
-        assert!(is_empty(&v), 0);
-        let BigVector { buckets, next_index: _, bucket_size: _ } = v;
+        assert!(is_empty(&v), Errors::invalid_argument(ENOT_EMPTY));
```

Comment:
should it be?  since there could v with empty buckets that can be deleted
shrink_to_fit(&mut v);        
assert!(is_empty(&v), Errors::invalid_argument(ENOT_EMPTY));

Reply:
is_empty only checks the actual items, but not the empty buckets so the order doesn't matter